indico/modules/events/registration/controllers/display.py

```diff
@@ -240,6 +253,32 @@ def _process(self):
         )
 
 
+class RHParticipantListPictureDownload(RHParticipantList):
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.registration
+        }
+    }
+
+    def _check_access(self):
+        RHParticipantList._check_access(self)
+        is_participant = self.registration.event.is_user_registered(session.user)
+        if not self.registration.is_publishable(is_participant):
```

Comment:
You also need to check if the picture is enabled in the participant list config. If it isn't, the picture must not be accessible via this endpoint even if the registration itself is visible in the participant list.

Reply:
Done.